polus-feature-subsetting-plugin/src/main.py

```diff
@@ -0,0 +1,257 @@
+import argparse, logging, subprocess, time, multiprocessing, sys
+import os
+import filepattern
+import pandas as pd
+import shutil
+from pathlib import Path
+import traceback
+
+def filter_planes(feature_dict, removeDirection, percentile):
+    """filter planes by the criteria specified by removeDirection
+    and percentile 
+
+    Args:
+        feature_dict (dictionary): planes and respective feature value
+        removeDirection (string): remove above or below percentile
+        percentile (int): cutoff percentile
+
+    Returns:
+        set: planes that fit the criteria
+    """
+    planes = list(feature_dict.keys())
+    feat_value = [feature_dict[i] for i in planes]
+    thresh = min(feat_value) + percentile * (max(feat_value) - min(feat_value))
+    
+    # filter planes
+    if removeDirection == 'Below':
+        keep_planes = [z for z in planes if feature_dict[z] >= thresh]
+    else:
+        keep_planes = [z for z in planes if feature_dict[z] <= thresh]
+    
+    return set(keep_planes)
+
+def make_uniform(planes_dict, uniques, delay):
+    """ this function makes the output collection uniform in
```

Comment:
It may not be a bad idea to try researching VS Codes ruler feature. The Python standard is that line widths shouldn't be more than 80 characters in general, and VS code can display rules for 80 characters. It helps to create more uniform code and documentation:
https://stackoverflow.com/q/29968499